protocol/messenger_community_storenodes.go

```diff
@@ -0,0 +1,82 @@
+package protocol
+
+import (
+	"context"
+
+	"github.com/golang/protobuf/proto"
+	"go.uber.org/zap"
+
+	"github.com/status-im/status-go/eth-node/crypto"
+	"github.com/status-im/status-go/eth-node/types"
+	"github.com/status-im/status-go/protocol/common"
+	"github.com/status-im/status-go/protocol/common/shard"
+	"github.com/status-im/status-go/protocol/communities"
+	"github.com/status-im/status-go/protocol/protobuf"
+	"github.com/status-im/status-go/protocol/storenodes"
+	v1protocol "github.com/status-im/status-go/protocol/v1"
+)
+
+func (m *Messenger) sendPublicSyncCommunityStorenodes(community *communities.Community, snodes storenodes.Storenodes) error {
+	if !community.IsControlNode() {
+		return communities.ErrNotControlNode
+	}
+
+	clock, chat := m.getLastClockWithRelatedChat()
+	pb := &protobuf.CommunityStorenodes{
+		Clock:       clock,
+		CommunityId: community.ID(),
+		Storenodes:  snodes.ToProtobuf(),
+		ChainId:     communities.CommunityDescriptionTokenOwnerChainID(community.Description()),
+	}
+	snPayload, err := proto.Marshal(pb)
+	if err != nil {
+		return err
+	}
+	signature, err := crypto.Sign(crypto.Keccak256(snPayload), community.PrivateKey())
+	if err != nil {
+		return err
+	}
+	signedSyncCommunityStorenodes := &protobuf.SyncCommunityStorenodes{
+		Signature: signature,
+		Payload:   snPayload,
+	}
+	signedPayload, err := proto.Marshal(signedSyncCommunityStorenodes)
+	if err != nil {
+		return err
+	}
+
+	rawMessage := common.RawMessage{
+		Payload:             signedPayload,
+		Sender:              community.PrivateKey(),
+		SkipEncryptionLayer: true,
+		MessageType:         protobuf.ApplicationMetadataMessage_SYNC_COMMUNITY_STORENODES,
+		PubsubTopic:         shard.DefaultNonProtectedPubsubTopic(),
+	}
+
+	_, err = m.sender.SendPublic(context.Background(), chat.Name, rawMessage)
```

Comment:
I believe this message should be sent not to `chat.Name`, but smth similar to `CommunityShardInfoTopic`, e.g. `<community-id>-store-nodes-info`.